common/psbt_open.c

```diff
@@ -434,6 +435,32 @@ int psbt_find_serial_output(struct wally_psbt *psbt, u16 serial_id)
 	return -1;
 }
 
+u16 psbt_new_input_serial(struct wally_psbt *psbt, enum tx_role role)
+{
+	const u64 serial_space = 10000000;
+	u16 serial_id;
+
+	while ((serial_id = pseudorand(serial_space)) % 2 != role ||
+		psbt_find_serial_input(psbt, serial_id) != -1) {
+		/* keep going; */
+	}
+
+	return serial_id;
```

Comment:
How about `(pseudorand(1 << 15) << 1) | role`?  Yeah, we still need to loop, and if will get inefficient once we've got about 65,000 inputs, but I think that's impossible anyway.

Reply:
Even better, make function `get_random_serial(role)` and reuse below...